bi_sql_editor/models/bi_sql_view.py

```diff
@@ -490,9 +507,14 @@ def _prepare_action_name(self):
 
     def _prepare_menu(self):
         self.ensure_one()
+        parent_id = (
+            self.parent_menu_id.id
+            or self.env.ref("bi_sql_editor.menu_bi_sql_editor").id
+        )
+        self.write({"parent_menu_id": parent_id})
```

Comment:
a write in a prepare is not the best design IMO.

I guess we could simply remove that lines, and set 

```
parent_menu_id = fields.Many2one(
    ...
    required=True,
    default=lambda x: x._default_parent_menu_id(),
    help="The parent menu on which the menu related to "
        "the SQL report will be created."
    ...
)

def _default_parent_menu_id(self):
    return self.env.ref("bi_sql_editor.menu_bi_sql_editor")

```

This will be more explicit.

What do you think ?

Reply:
Great, I totally agree that this looks like a cleaner solution. I will update 👍🏿